### PR TITLE
Change library location (bin/{cpu}/{os_bin}) + Expand CI targets

### PR DESCRIPTION
**Pull request by alextwoods**
*Description of changes:*
* Change library location (bin/{cpu}/{os_bin}).  This should make it a bit more straightforward to add/support different platforms (combinations of cpu + os).  
* Add windows targets to CI  [Dropped linux, have been unable to get this to work at all]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: builder.json [hunk 1]
```diff
@@ -3,6 +3,19 @@
     "packages": [
         "ruby"
     ],
+    "targets": {
+        "linux": {
+            "variables": {
+                "openssl_include": "/opt/openssl/include",
+                "openssl_lib": "/opt/openssl/lib"
```
Comment (graebm) [inline builder.json:10]: ohhh yeah this weird crap
we need a gem/rake idiomatic way for the user to pass info all the way down to cmake
feel free to check this in as-is, or revert this file entirely
we can fight linux together tomorrow

**Review by graebm — APPROVED** [review]

File: aws-crt/ext/extconf.rb [hunk 2]
```diff
@@ -18,9 +18,9 @@
...
 
   platform = local_platform
-  binary_name = PLATFORM_BINARIES[platform]
-  src_name = "native/build/#{binary_name}"
-  dest_name = "bin/#{platform}/#{binary_name}"
-  FileUtils.mkdir_p("bin/#{platform}")
+  binary_name = crt_bin_name(platform)
+  src_name = PLATFORM_BUILD_PATHS[platform] || "native/build/#{binary_name}"
+  dest_name = "bin/#{platform.cpu}/#{binary_name}"
+  FileUtils.mkdir_p("bin/#{platform.cpu}")
```
[the remark this answers is not in the document]
Comment (alextwoods) [reply aws-crt/ext/extconf.rb:24]: This is used for pure ruby gems - its the only hook for a "post install" step - it does need to generate a blank makefile (which is done above this), but it also compiles and moves the crt lib to bin/

I'll add comments to clarify this.

File: builder.json (hunk 1 above)
Comment (alextwoods) [reply builder.json:10]: Yeah - I'll remove this for now.  Windows is working with minimal weirdness. Linux has more issues than just this...

File: aws-crt/lib/aws-crt/platforms.rb [hunk 3]
```diff
@@ -1,28 +1,34 @@
...
+OS_BINARIES = {
+  'darwin' => 'libaws-crt.dylib',
+  'linux' => 'libaws-crt.so',
+  'mingw32' => 'aws-crt.dll'
+}.freeze
+
+DEFAULT_BINARY = 'libaws-crt.so'.freeze
+
+PLATFORM_BUILD_PATHS = {
+  'x86_64-mingw32' => 'native/build/Debug/aws-crt.dll'
```
[the remark this answers is not in the document]
Comment (alextwoods) [reply aws-crt/lib/aws-crt/platforms.rb:13]: Yeah - I messed around a little but couldn't get anything better working, but I don't have any experience with cmake.

**Pull request merged**